src/Compatibility/Microsoft.DotNet.ApiCompatibility/Logging/SuppressionEngine.cs

```diff
@@ -22,16 +22,19 @@ public class SuppressionEngine
         protected HashSet<Suppression> _validationSuppressions;
         private readonly ReaderWriterLockSlim _readerWriterLock = new();
         private readonly XmlSerializer _serializer = new(typeof(Suppression[]), new XmlRootAttribute("Suppressions"));
-        private readonly HashSet<string?> _noWarn;
+        private readonly HashSet<string> _noWarn;
 
-        protected SuppressionEngine(string? suppressionFile, string noWarn)
+        /// <summary>
+        /// Creates a new instance of <see cref="SuppressionEngine"/>.
+        /// </summary>
+        /// <param name="noWarn"></param>
+        /// <param name="suppressionFile">The path to the suppressions file to be used for initialization.</param>
+        public SuppressionEngine(string? noWarn = null, string? suppressionFile = null)
```

Comment:
Seems like you are reversing the order of the params and params are both strings so we should be very careful on all of the callsites and make sure they all got updated correctly. Additionally, if there is no need to update the ordering, leaving it how it was is preferrable.